source/_posts/2018-12-05-swift-alps-2018.md

```diff
@@ -0,0 +1,178 @@
+
+The goal was to continue the work started by Florent on either an iOS chat client using Network.framework, or a macOS chat server using Swift NIO, and my session buddy and I chose to start out with the client app.
+
+In my normal, day to day work, networking work is something you set up at the start of a project - typically by coping some code you were satisfied with in a previous project, and then its just there, calling services, translating the returned JSON into model objects that you can use for your "real" work.
+
+Therefore it was really fun and challenging to try and wrap your head around implementing a chat client from scratch. Starting out with making a connection (deciding whether to use TCP or UDP), actually connecting and watch the connection go through different stages before - hopefully - being established, and finally being able to send data over the network.
+
+Apples new [Network.framework](https://developer.apple.com/documentation/network) was pleasant to work with, it felt very swifty with various handler closures and enums with associated values.
+
+An interesting feature in Network.framework is the [NWPath](https://developer.apple.com/documentation/network/nwpath) and [NWPathMonitor](https://developer.apple.com/documentation/network/nwpathmonitor) which you can use to monitor the connection and react if it changes...or completely goes away. Super interesting when writing a network monitor for instance.
+
+All in all a very good workshop, good work Florent
+
+## An "Intents" Guide to Custom Siri Shortcuts
+
+Siri shortcuts were one of the bigger announcements to come out of WWDC 18. There's a lot of talk about powerful workflows you can add to your app to make your user experience better for both casual and power users alike. In Aleksander's workshop, we learned about how siri shortcuts work, or specifically the Intents framework works, and then implemented them in existing apps we had. We soon found out two important facts about developing siri shortcuts.
+
+1. Creating a new shortcut is super simple. All you need to do is create a new intent shortcut, provide it some info about what data it is expecting, what it should listen for, and what kind of action it is (voice call, ordering food, playing music, etc). Build it and run and you should be able to get siri to communicate with your app right away.
+
+2. Structuring your app to perform useful tasks with the data input from siri is a huge complex problem. Since the intent is in its own extension, it does not have access to the rest of your codebase in the traditional sense. You need to modularize your code in a way that separates logic into separate services, which can then be imported into the intent extension. If your app isn't already set up like this, it can be a huge refactor effort.

```

Comment:
`separates logic into separate parts` sounds repetitive, maybe use a different term in one of the cases?

Reply:
huge `refactoring` effort maybe?

Reply:
> You need to modularize your code in a way that splits logic into separate services